src/SecurityManager.ts

```diff
@@ -98,11 +98,24 @@ async function getSecretStorageKey(
     { keys: keyInfos }: { keys: Record<string, ISecretStorageKeyInfo> },
     ssssItemName,
 ): Promise<[string, Uint8Array]> {
-    const keyInfoEntries = Object.entries(keyInfos);
-    if (keyInfoEntries.length > 1) {
-        throw new Error("Multiple storage key requests not implemented");
+    const cli = MatrixClientPeg.get();
+    let keyId = await cli.getDefaultSecretStorageKeyId();
+    let keyInfo;
+    if (keyId) {
+        // use the default SSSS key if set
+        keyInfo = keyInfos[keyId];
+        if (!keyInfo) {
```

Comment:
I am worried that throwing an error here (when the default is not present in the infos presented) may break more complex scenarios, such as someone who had 4S setup, but then resets. They may then have at least 2 secret storage keys, and may need to read (at least for a time) things encrypted with non-default storage keys. In our current process, there's always one key marked as the default, but it could be the wrong one for the thing you want to read.

Can we rework this to attempt the default key, but if that's not in the provided set, then fall back to the previous mode of picking the only provided key?

Reply:
fixed